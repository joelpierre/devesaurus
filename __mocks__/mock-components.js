export const mockComponent = {
  component: {
    acf_fc_layout: 'cta_banner',
    heading: 'Get in touch',
    copy: 'Why not get in touch today',
    cta: true,
    cta_text: 'Some text',
    cta_link: {
      ID: 187,
      post_author: '1',
      post_date: '2019-03-08 20:42:03',
      post_date_gmt: '2019-03-08 20:42:03',
      post_content: '',
      post_title: 'Terms and Conditions',
      post_excerpt: '',
      post_status: 'publish',
      comment_status: 'closed',
      ping_status: 'closed',
      post_password: '',
      post_name: 'terms-and-conditions',
      to_ping: '',
      pinged: '',
      post_modified: '2019-03-13 12:58:02',
      post_modified_gmt: '2019-03-13 12:58:02',
      post_content_filtered: '',
      post_parent: 0,
      guid: 'http://devesaurus.test/?page_id=187',
      menu_order: 0,
      post_type: 'page',
      post_mime_type: '',
      comment_count: '0',
      filter: 'raw',
    },
    cta_theme: 'alpha',
    theme: 'brand',
  },
};

export const mockComponents = [mockComponent];
