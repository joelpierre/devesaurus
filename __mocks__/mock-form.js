export const mockFormGroups = [
  {
    fieldSet: {
      legend: 'FormBase Legend',
      description: 'some text here',
      fields: [
        {
          type: 'input',
          columns: {
            colMd: 6,
          },
          input: {
            label: 'label-1',
            name: 'test-1',
            type: 'text',
            placeholder: 'placeholder-1',
          },
        },
        {
          type: 'select',
          columns: {
            colMd: 6,
          },
          input: {
            label: 'label-2',
            name: 'test-2',
            placeholder: 'placeholder-1',
            options: [
              {
                name: 'option-name',
                value: 'option-value',
              },
            ],
          },
        },
        {
          type: 'file',
          columns: {},
          input: {
            name: 'file-name',
            placeholder: 'file-placeholder',
          },
        },
      ],
    },
  },
];

export const mockFormData = {
  formName: 'test Search form',
  buttons: [
    {
      type: 'submit',
      text: 'Submit Form',
      theme: 'brand',
    },
    {
      type: 'reset',
      text: 'Reset Form',
      theme: 'beta',
    },
  ],
  formGroups: mockFormGroups,
};
