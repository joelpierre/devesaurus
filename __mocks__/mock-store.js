import configureStore from 'redux-mock-store';

export const MockMiddleware = [];

export const MockState = {
  core: {
    options: {
      company_name: 'Testing',
      company_slogan: 'Testing Slogan',
    },
  },
};

export const MockStoreSteup = configureStore(MockMiddleware);

export const MockStore = MockStoreSteup(MockState);
