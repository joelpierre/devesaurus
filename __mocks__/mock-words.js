export const mockWord = {
  title: 'Graphic Design',
  slug: 'graphic-design',
  terms: [
    {
      name: 'Design',
      slug: 'design',
      taxonomy: 'word_tag',
    },
    {
      name: 'Graphic Design',
      slug: 'graphic-design',
      taxonomy: 'word_category',
    },
  ],
  acf: {
    definition: '<p>The art or skill of combining text and pictures in advertisements, magazines, or books.</p>\n',
    origin: {
      value: 'GB',
      label: 'United Kingdom',
    },
    syllables: {
      count: '4',
      list: [
        {
          item: 'gra',
        },
        {
          item: 'phic',
        },
        {
          item: 'de',
        },
        {
          item: 'sign',
        },
      ],
    },
    pronunciation: 'graphic-de-sign',
    part_of_speech: 'noun',
    synonyms: [
      {
        post_title: 'Web Development',
        post_status: 'publish',
        post_name: 'web-development',
      },
    ],
    examples: false,
  },
};

export const mockWords = [
  mockWord,
  {
    title: 'Web Development',
    slug: 'web-development',
    terms: [
      {
        name: 'CSS',
        slug: 'css',
        taxonomy: 'word_tag',
      },
      {
        name: 'HTML5',
        slug: 'html5',
        taxonomy: 'word_tag',
      },
      {
        name: 'Web Development',
        slug: 'web-development',
        taxonomy: 'word_category',
      },
    ],
    acf: {
      definition: '<p>Web development is the work involved in developing a web site for the Internet (World Wide Web) or an intranet (a private network). Among web professionals, "web development" usually refers to the main non-design aspects of building web sites: writing markup and coding</p>\n',
      origin: {
        value: 'GB',
        label: 'United Kingdom',
      },
      syllables: {
        count: '5',
        list: [
          {
            item: 'web',
          },
          {
            item: 'de',
          },
          {
            item: 've',
          },
          {
            item: 'lop',
          },
          {
            item: 'ment',
          },
        ],
      },
      pronunciation: 'web~develop~ment',
      part_of_speech: 'noun',
      synonyms: [
        {
          post_title: 'Graphic Design',
          post_status: 'publish',
          post_name: 'graphic-design',
        },
      ],
      examples: [
        {
          type: 'code-block',
          codepen_url: '',
          code: '<p><code></p>\n<header>\ntest<br />\n</header>\n<p></code></p>\n',
        },
      ],
    },
  },
];
