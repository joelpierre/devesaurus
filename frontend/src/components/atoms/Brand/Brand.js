import React from 'react';
import * as styles from './Brand.module.scss';
import SiteInfo from '../../../hoc/siteInfo';

function Brand({ classes }) {
  return (
    <div className={`${classes}`}>
      <div className={`${styles.brand}`}>
        <h1 className={styles.brand__heading}>
          <SiteInfo />
        </h1>
      </div>
    </div>
  );
}

export default Brand;
