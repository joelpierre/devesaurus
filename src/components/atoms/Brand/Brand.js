import React from 'react';
import { connect } from 'react-redux';
import * as styles from './Brand.module.scss';

function Brand({ classes, siteOptions }) {
  return (
    <div className={`${classes}`}>
      <div className={`${styles.brand}`}>
        <h1 className={styles.brand__heading}>
          {siteOptions.company_name}
        </h1>
      </div>
    </div>
  );
}

const mapStateToProps = state => ({
  siteOptions: state.core.options,
});

export default connect(mapStateToProps)(Brand);
