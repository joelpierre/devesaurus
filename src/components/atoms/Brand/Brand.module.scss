@import "~sass/core";

.brand {
  $root: &;
  width: 200px;
  height: 60px;
  padding: 0 10px;
  box-sizing: content-box;
  overflow: hidden;
  position: relative;

  @include tablet-landscape {
    width: 300px;
  }

  &__wrapper {
    width: 100%;
    height: 100%;
    display: flex;
    align-items: flex-start;
  }

  &__company {
    @include sr-only;
  }

  &__icon,
  &__logo {
    width: 100%;
    height: auto;
  }


  &--left {
    margin-left: 0;
    margin-right: auto;

    #{$root} {
      &__wrapper {
        width: auto;
      }

      &__icon,
      &__logo {
        width: auto;
        margin-right: auto;
        margin-left: 0;
      }
    }
  }

  &--right {
    #{$root} {
      &__wrapper {
        width: auto;
      }

      &__icon,
      &__logo {
        margin-left: auto;
        margin-right: 0;
      }
    }
  }

  &--center {
    margin: 0 auto;

    #{$root} {
      &__wrapper {
        width: 100%;
      }

      &__icon,
      &__logo {
        margin-right: auto;
        margin-left: auto;
      }
    }
  }
}
