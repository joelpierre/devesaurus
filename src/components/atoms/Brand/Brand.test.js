import React from 'react';
import renderer from 'react-test-renderer';
import Brand from './Brand';
import { Provider } from 'react-redux';
import { MockStore } from '../../../../__mocks__/mock-store';


describe('<Brand/>', () => {
  it('Should render Brand component', () => {
    const component = renderer.create(
      <Provider store={MockStore}>
        <Brand/>
      </Provider>)
      .toJSON();

    expect(component)
      .toMatchSnapshot();
  });
});
