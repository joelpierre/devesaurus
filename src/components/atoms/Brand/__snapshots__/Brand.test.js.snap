// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brand/> Should render Brand component 1`] = `
<div
  className="undefined"
>
  <div
    className="undefined"
  >
    <h1>
      Testing
    </h1>
  </div>
</div>
`;
